src/main.cpp

```diff
@@ -4399,6 +4400,37 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         else
             pfrom->fRelayTxes = true;
 
+
+        // FIXME: Overwinter expects a branch id
+        int height = GetHeight();
+        if (NetworkUpgradeActive(height, chainparams.GetConsensus(), Consensus::UPGRADE_TESTDUMMY)) {
+
+            // Catch compile-time error with version constants.  This code requires protocol version >= OVERWINTER_PROTO_VERSION
+            static_assert(PROTOCOL_VERSION >= OVERWINTER_PROTO_VERSION, "protocol version must be >= Overwinter protocol version");
+
+            if (!vRecv.empty()) {
+                vRecv >> pfrom->nBranchId;
+
+                // FIXME: Is this branch ID acceptable to this node?  Set of acceptable branchIDs?
+                NUInfo info = NetworkUpgradeInfo[Consensus::UPGRADE_TESTDUMMY];
```

Comment:
Oh, I completely forgot that the original intent of this field was to disambiguate the protocol version, which adds another option:

4. Branch ID corresponding to the branch this protocol version was introduced for (Overwinter) or during (Sprout).

However, thinking about this more, we never actually parse data conditional on the protocol version in the handshake (since we could be talking to a supportable older node). So we might need two (or three) things:

- Protocol format branch ID
- Peer's consensus branch ID (one of the three options above)
- Possibly something else to enable reasonable versioning/parsing of the handshake message itself (but this might be satisfied by the protocol format branch ID).

Reply:
@str4d the protocol version branch id - is this the branch id when the protocol version first goes live?